src/types.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright 2016 The Incremental DOM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/** @typedef {string|function(new: Element): Element} */
```

Comment:
We had to do some [jiggery](https://github.com/ampproject/amphtml/pull/5243/files#diff-8e1cdba946c07a7e631ef74645a5d6f0R381) to get Custom Elements to work around a [bug](https://github.com/babel/babel/issues/4480) with the v1 spec and ES6 classes. Specifically, this should be function that take and return an `Element` instance.